micro-rdk/src/common/ota.rs

```diff
@@ -304,15 +304,8 @@ impl<S: OtaMetadataStorage> OtaService<S> {
         self.stored_metadata().unwrap_or_default().version != self.pending_version
     }
 
-    /// Attempts to perform an OTA update.
-    /// On success, returns an `Ok(true)` or `Ok(false)` indicating if a reboot is necessary.
-    pub(crate) async fn update(&mut self) -> Result<bool, OtaError<S>> {
-        if !(self.needs_update()) {
-            return Ok(false);
-        }
-
-        let mut uri = self
-            .url
+    fn parse_uri(&self, url: &str) -> Result<hyper::Uri, OtaError<S>> {
```

Comment:
I considered using `fn parse_uri(&mut self)` and setting the `url` within the `OtaService` struct itself, but figured keeping track of the original url may still be useful for debugging later. Though the original can also be found in the robot config. happy to change it if there's a preference

Reply:
I think the function is fine as is.